src/meep.hpp

```diff
@@ -1188,6 +1190,25 @@ class dft_force {
   volume where;
 };
 
+
+
+struct near_data{
+  vec near_x;
+  component near_fd_comp;
+  std::vector<std::vector<std::complex<double> > > matrix_elt;
+};
+
+struct sourcedata{
+  component near_fd_comp;
+  std::vector<ptrdiff_t> idx_arr;
+  int fc_idx;
+  std::vector<std::vector<std::complex<double> > > amp_arr;
```

Comment:
should be 1d array in row-major

Reply:
(storing multidimensional arrays as arrays of arrays will be much harder to get into Python — numpy arrays are just "flat" row-major arrays)